sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -16281,6 +16281,71 @@ public void testTimeStampAddConversion() throws Exception
     );
   }
 
+  @Test
+  public void testGroupingSetsWithLimit() throws Exception
```

Comment:
Skimming through the code - I think we should add some tests to test the iterators built for a query with an `ORDER BY` on dimensions that are part of the grouping sets (like `dim2`) and for something that isn't (like `SUM(cnt)` or another dimension)